Python/ceval.c

```diff
@@ -5998,3 +6007,38 @@ void Py_LeaveRecursiveCall(void)
 {
     _Py_LeaveRecursiveCall_inline();
 }
+
+static void improve_missing_self_error(PyThreadState* tstate, PyCodeObject* co, Py_ssize_t nargs) {
+
+    if (nargs + 1 != co->co_argcount) {
```

Comment:
What about positional-only parameters and var-positional parameter?

Reply:
I left that for future PRs, I wanted to get the basic logic first and then improve upon

Reply:
There is the check `!(co->co_flags & CO_VARARGS)` befory calling `too_many_positional()`.